tools/compogen/pkg/gen/readme.go

```diff
@@ -356,3 +513,71 @@ func firstToLower(s string) string {
 
 	return string(mod) + s[size:]
 }
+
+func enumValues(enum []string) string {
+	length := len(enum)
+	var result string
+	if length <= 5 {
+		result = "<br/>Enum: <br/><ul>"
+	} else {
+		result = "<br/><details> <summary>Enum: </summary><ul>"
+	}
+	for i, e := range enum {
+		result += fmt.Sprintf("<li>%s</li>", e)
+		if i == length-1 {
+			result += "</ul>"
+		}
+	}
+	if length > 5 {
+		result += "</details>"
+	}
+	return result
+}
+
+func firstToCapital(s string) string {
+	r, size := utf8.DecodeRuneInString(s)
+	if r == utf8.RuneError && size <= 1 {
+		return s
+	}
+
+	mod := unicode.ToUpper(r)
+	if r == mod {
+		return s
+	}
+
+	return string(mod) + s[size:]
+}
+
+func findConstantValue(option objectSchema) string {
+	for _, prop := range option.Properties {
+		if prop.Const != "" {
+			return prop.Const
+		}
+	}
+	return ""
+}
+
+func AnchorObjectTitle(p interface{}) string {
```

Comment:
I will think about how to make anchor better. I may use some identifiers that won't be seen by users.

Reply:
> Response-format is an input property, too, but it has a section outside of "Input Objects"

Yes, it is a good point. Though they are both "Input Object", in details, they are with different attributes.
So, it is the reason why I design the code like it. 
I imagine (feeling) it could benefit us by differentiating them.
<img width="359" alt="image" src="https://github.com/user-attachments/assets/1e4a51da-73a9-456b-a8f3-4c2b6bc5b982">

Reply:
> Links don't work when the target is collapsed

It seems it does not work in GitHub & VScode Preview, but it works in instill.tech page.

Reply:
> Make the anchor ID as specific as possible to avoid collisions across tasks, input/output, oneOf, etc. I don't know the sweet spot here, to be unique we'd need to include task, input/output and the object tree, but that might result in anchors that are too long.

Now, I use task-name-prop-key.
It'd be a bit hard to do
1. trace the tree based on the object key.
2. add the identifier for the anchor text and id together.

To achieve that, it will require more time to develop it.
Now, it seems there is only GCS meet collisions with object Attribute.
So, I want to improve that part later.

Reply:
Now, I use task-name-prop-key.

It'd be a bit hard to do
1. trace the tree based on the object key.
2. add the identifier for the anchor text and id together.

To achieve that, it will require more time to develop it.

Now, it seems there is only GCS meet collisions with object Attribute.
So, I want to improve that part later.

Reply:
> It seems it does not work in GitHub & VScode Preview, but it works in instill.tech page.

Hm 🤔 not on Firefox, but at least it isn't something happening everywhere. Nice to have if we could have a consistent experience